File: EX1/EventMasterSoft/EventMasterSoftDA/MySQL/EventoMySQL.cs
using EventMasterSoftDA.DAO;
using EventMasterSoftDBManager;
using EventMasterSoftModel;
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EventMasterSoftDA.MySQL
{
    public class EventoMySQL : EventoDAO
    {
        private MySqlDataReader lector;

        public int insertar(Evento evento)
        {
            int resultado = 0;
            MySqlParameter[] parametros = new MySqlParameter[11];
            parametros[0] = new MySqlParameter("_id_evento", MySqlDbType.Int32);
            parametros[0].Direction = System.Data.ParameterDirection.Output;
            parametros[1] = new MySqlParameter("_fid_productora", evento.Productora.IdProductora);
            parametros[2] = new MySqlParameter("_fid_clasificacion", evento.Clasificacion);
            parametros[3] = new MySqlParameter("_nombre", evento.Nombre);
            parametros[4] = new MySqlParameter("_costo_realizacion", evento.CostoRealizacion);
            parametros[5] = new MySqlParameter("_tipo_evento", evento.TipoEvento.ToString());
            parametros[6] = new MySqlParameter("_fecha_realizacion", evento.FechaRealizacion);
            parametros[7] = new MySqlParameter("_descripcion", evento.Descripcion);
            parametros[8] = new MySqlParameter("_permite_reingreso", evento.PermiteReingreso);
            parametros[9] = new MySqlParameter("_permite_grabacion", evento.PermiteGrabacion);
            parametros[10] = new MySqlParameter("_banner_promocional", evento.BannerPromocional);
            evento.IdEvento = DBManager.Instance.EjecutarProcedimiento("INSERTAR_EVENTO", parametros, "_id_evento");
            resultado = evento.IdEvento;
            return resultado;
        }

        public BindingList<Evento> listarPorNombre(string nombre)
        {
            BindingList<Evento> eventos = new BindingList<Evento>();
            MySqlParameter[] parametros = new MySqlParameter[1];
            parametros[0] = new MySqlParameter("_nombre", nombre);
            lector = DBManager.Instance.EjecutarProcedimientoLectura("LISTAR_EVENTOS_X_NOMBRE", parametros);
            while (lector.Read())
            {
                Evento evento = new Evento();

                evento.IdEvento = lector.GetInt32("id_evento");
                evento.Productora = new Productora();
                evento.Productora.IdProductora = lector.GetInt32("id_productora");
                evento.Productora.Nombre = lector.GetString("nombre_productora");
                evento.Nombre = lector.GetString("nombre_evento");
                evento.FechaRealizacion = lector.GetDateTime("fecha_realizacion");

                eventos.Add(evento);
            }
            DBManager.Instance.CerrarConexion();
            return eventos;
        }

        public Evento obtenerPorId(int idEvento)
        {
            Evento evento = new Evento();
            MySqlParameter[] parametros = new MySqlParameter[1];
            parametros[0] = new MySqlParameter("_id_evento", idEvento);
            lector = DBManager.Instance.EjecutarProcedimientoLectura("OBTENER_EVENTO_X_ID", parametros);
            while (lector.Read())
            {
                evento.IdEvento = lector.GetInt32("id_evento");
                evento.Productora = new Productora();
                evento.Productora.IdProductora = lector.GetInt32("id_productora");
                evento.Productora.Nombre = lector.GetString("nombre_productora");
                evento.Clasificacion = lector.GetChar("id_clasificacion");
                evento.Nombre = lector.GetString("nombre_evento");
                evento.CostoRealizacion = lector.GetDouble("costo_realizacion");
                evento.TipoEvento = (TipoEvento)Enum.Parse(typeof(TipoEvento), lector.GetString("tipo_evento"));
                evento.FechaRealizacion = lector.GetDateTime("fecha_realizacion");
                evento.Descripcion = lector.GetString("descripcion");
                evento.PermiteReingreso = lector.GetBoolean("permite_reingreso");
                evento.PermiteGrabacion = lector.GetBoolean("permite_grabacion");
                evento.BannerPromocional = (byte[])lector["banner_promocional"];
                evento.Activo = lector.GetBoolean("activo");
            }
            DBManager.Instance.CerrarConexion();
            return evento;
        }
    }
}

File: LAB5/Backend/EventMasterSoftDA/src/pe/edu/pucp/eventmastersoft/mysql/EventoMySQL.java
package pe.edu.pucp.eventmastersoft.mysql;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.HashMap;
import pe.edu.pucp.eventmastersoft.config.DBManager;
import pe.edu.pucp.eventmastersoft.dao.EventoDAO;
import pe.edu.pucp.eventmastersoft.model.Evento;
import pe.edu.pucp.eventmastersoft.model.Productora;
import pe.edu.pucp.eventmastersoft.model.TipoEvento;
import java.sql.Types;

public class EventoMySQL implements EventoDAO{

    private ResultSet rs;

    @Override
    public int insertar(Evento evento) {
        HashMap<String,Object> parametrosEntrada = new HashMap<>();
		parametrosEntrada.put("_fid_productora", evento.getProductora().getIdProductora());
		parametrosEntrada.put("_fid_clasificacion", evento.getClasificacion());
		parametrosEntrada.put("_nombre", evento.getNombre());
		parametrosEntrada.put("_costo_realizacion", evento.getCostoRealizacion());
		parametrosEntrada.put("_tipo_evento", evento.getTipoEvento());
		parametrosEntrada.put("_fecha_realizacion", evento.getFechaRealizacion());
		parametrosEntrada.put("_descripcion", evento.getDescripcion());
		parametrosEntrada.put("_permite_reingreso", evento.isPermiteReingreso());
		parametrosEntrada.put("_permite_grabacion", evento.isPermiteGrabacion());
		parametrosEntrada.put("_banner_promocional", evento.getBannerPromocional());

		HashMap<String,Object> parametrosSalida = new HashMap<>();
		parametrosSalida.put("_id_evento", Types.INTEGER);

		DBManager.getInstance().ejecutarProcedimiento("INSERTAR_EVENTO", parametrosEntrada, parametrosSalida);
        return (int) parametrosSalida.get("_id_evento");
    }

    @Override
    public ArrayList<Evento> listarPorNombre(String nombre) {
        ArrayList<Evento> eventos = new ArrayList<>();
		HashMap<String,Object> parametrosEntrada = new HashMap<>();
		parametrosEntrada.put("_nombre", nombre);
		rs = DBManager.getInstance().ejecutarProcedimientoLectura("LISTAR_EVENTOS_X_NOMBRE", parametrosEntrada);
		try{
			while(rs.next()){
			   Evento evento = new Evento();
			   evento.setIdEvento(rs.getInt("id_evento"));
			   evento.setNombre(rs.getString("nombre_evento"));
			   evento.setFechaRealizacion(rs.getDate("fecha_realizacion"));
			   Productora productora = new Productora();
			   productora.setIdProductora(rs.getInt("id_productora"));
			   productora.setNombre(rs.getString("nombre_productora"));
			   evento.setProductora(productora);
			   eventos.add(evento);
			}
		}catch(SQLException ex){
			System.out.println("Error leyendo datos: " + ex.getMessage());
		}
		return eventos;
    }

    @Override
    public Evento obtenerPorId(int idEvento) {
        Evento evento = new Evento();
		HashMap<String,Object> parametrosEntrada = new HashMap<>();
		parametrosEntrada.put("_id_evento", idEvento);
		rs = DBManager.getInstance().ejecutarProcedimientoLectura("OBTENER_EVENTO_X_ID", parametrosEntrada);
		try{
			if(rs.next()){
			   evento.setIdEvento(rs.getInt("id_evento"));
			   evento.setClasificacion(rs.getString("id_clasificacion").charAt(0));
			   evento.setCostoRealizacion(rs.getDouble("costo_realizacion"));
			   evento.setTipoEvento(TipoEvento.valueOf(rs.getString("tipo_evento")));
			   evento.setNombre(rs.getString("nombre_evento"));
			   evento.setDescripcion(rs.getString("descripcion"));
			   evento.setBannerPromocional(rs.getBytes("banner_promocional"));
			   evento.setPermiteGrabacion(rs.getBoolean("permite_grabacion"));
			   evento.setPermiteReingreso(rs.getBoolean("permite_reingreso"));
			   evento.setFechaRealizacion(rs.getDate("fecha_realizacion"));
			   Productora productora = new Productora();
			   productora.setIdProductora(rs.getInt("id_productora"));
			   productora.setNombre(rs.getString("nombre_productora"));
			   evento.setProductora(productora);
			   evento.setActivo(true);
			}
		}catch(SQLException ex){
			System.out.println("Error leyendo datos: " + ex.getMessage());
		}
		return evento;
    }
}
